Backend/app/alchemist/src/Main.hs

```diff
@@ -46,36 +50,73 @@ applyDirectory dirPath processFile = do
+  if exists
+    then do
+      hashObjectAtHEAD <- getHashObjectAtHEAD filePath
+      hashObject <- getHashObject filePath
+      return $
+        if isNothing hashObjectAtHEAD
+          then NEW
+          else
+            if hashObjectAtHEAD /= hashObject
+              then CHANGED
+              else UNCHANGED
+    else return NOT_EXIST
+
 main :: IO ()
 main = do
   putStrLn ("Version " ++ NammaDSL.version)
+  generateAllSpecs <- ("--all" `elem`) <$> getArgs
   currentDir <- getCurrentDirectory
   maybeGitRoot <- findGitRoot currentDir
   let rootDir = fromMaybe (error "Could not find git root") maybeGitRoot
 
-  processApp riderAppDatabaseName rootDir riderAppPath rideAppName
-  processApp driverAppDatabaseName rootDir driverAppPath driverAppName
+  processApp generateAllSpecs riderAppDatabaseName rootDir riderAppPath rideAppName
+  processApp generateAllSpecs driverAppDatabaseName rootDir driverAppPath driverAppName
   where
-    processApp :: String -> FilePath -> FilePath -> FilePath -> IO ()
-    processApp dbName rootDir appPath appName = do
-      applyDirectory (rootDir </> appPath </> "spec" </> "Storage") (processStorageDSL dbName rootDir appPath appName)
-      applyDirectory (rootDir </> appPath </> "spec" </> "API") (processAPIDSL rootDir appPath)
-
-    processStorageDSL dbName' rootDir appPath appName inputFile = do
-      let readOnlySrc = rootDir </> appPath </> "src-read-only/"
-      let src = rootDir </> appPath </> "src"
-      let readOnlyMigration = rootDir </> sqlOutputPathPrefix </> appName
-
-      NammaDSL.mkBeamTable (readOnlySrc </> "Storage/Beam") inputFile
-      NammaDSL.mkBeamQueries (readOnlySrc </> "Storage/Queries") (Just (src </> "Storage/Queries")) inputFile
-      NammaDSL.mkDomainType (readOnlySrc </> "Domain/Types") inputFile
-      NammaDSL.mkSQLFile (Just dbName') readOnlyMigration inputFile
-
-    processAPIDSL rootDir appPath inputFile = do
-      let readOnlySrc = rootDir </> appPath </> "src-read-only/"
-      let src = rootDir </> appPath </> "src"
-
-      -- NammaDSL.mkFrontendAPIIntegration (readOnlySrc </> "Domain/Action") inputFile
-      NammaDSL.mkServantAPI (readOnlySrc </> "API/Action/UI") inputFile
-      NammaDSL.mkApiTypes (readOnlySrc </> "API/Types/UI") inputFile
-      NammaDSL.mkDomainHandler (src </> "Domain/Action/UI") inputFile
+    processApp :: Bool -> String -> FilePath -> FilePath -> FilePath -> IO ()
+    processApp isGenAll dbName rootDir appPath appName = do
+      applyDirectory (rootDir </> appPath </> "spec" </> "Storage") (processStorageDSL isGenAll dbName rootDir appPath appName)
+      applyDirectory (rootDir </> appPath </> "spec" </> "API") (processAPIDSL isGenAll rootDir appPath)
+
+    processStorageDSL isGenAll dbName' rootDir appPath appName inputFile = do
+      fileState <- getFileState inputFile
+      putStrLn $ show fileState ++ " " ++ inputFile
+      when (isGenAll || fileState == NEW || fileState == CHANGED) $ do
```

Comment:
What about fileState doesn't exist?

Reply:
Does not matter as it will take provide only the existing files using this fuction : 
applyDirectory :: FilePath -> (FilePath -> IO ()) -> IO ()
applyDirectory dirPath processFile = do
  exists <- doesDirectoryExist dirPath
  when exists $ do
    files <- listDirectory dirPath.  <<<<<<<<<< This one 
    let yamlFiles = filter (\file -> takeExtension file `elem` [".yaml", ".yml"]) files
    mapM_ (processFile . (dirPath </>)) yamlFiles